Ginger/GingerCore/ALM/Qtest/ExportToQtest.cs

```diff
@@ -0,0 +1,594 @@
+
+                    //set the upload path
+                    TreeManager treeM = (TreeManager)mTDConn.TreeManager;
+                    ISysTreeNode testParentFolder = (ISysTreeNode)treeM.get_NodeByPath(uploadPath);
+                    test["TS_SUBJECT"] = testParentFolder.NodeID;
+                    
+                }
+                else
+                {
+                    //##update existing test case
+                    test = ImportFromQC.GetQCTest(activitiesGroup.ExternalID);
+
+                    //delete the un-needed steps
+                    DesignStepFactory stepF = test.DesignStepFactory;
+                    List stepsList = stepF.NewList("");
+                    foreach (DesignStep step in stepsList)
+                    {
+                        if (activitiesGroup.ActivitiesIdentifiers.Where(x => x.IdentifiedActivity.ExternalID == step.ID.ToString()).FirstOrDefault() == null)
+                            stepF.RemoveItem(step.ID);
+                    }
+
+                    //delete the existing parameters
+                    StepParams testParams = test.Params;
+                    if (testParams.Count > 0)
+                    {
+                        for (int indx = 0; indx < testParams.Count; indx++)
+                        {
+                            testParams.DeleteParam(testParams.ParamName[indx]);
+                            testParams.Save();
+                        }
+                    }
+
+                }
+
+                //set item fields
+                foreach (ExternalItemFieldBase field in testCaseFields)
+                {
+                    if (field.ToUpdate || field.Mandatory)
+                    {
+                        if (string.IsNullOrEmpty(field.SelectedValue) == false && field.SelectedValue != "NA")
+                            test[field.ID] = field.SelectedValue;
+                        else
+                            try { test[field.ID] = "NA"; }
+                            catch { }
+                    }
+                }
+                
+                //post the test
+                test.Name = activitiesGroup.Name;
+                test.Post();
+                activitiesGroup.ExternalID = test.ID.ToString();
+                activitiesGroup.ExternalID2 = test.ID.ToString();
+
+                //Add/update all test steps + Parameters
+                foreach (ActivityIdentifiers actIdent in activitiesGroup.ActivitiesIdentifiers)
+                    ExportActivityAsTestStep(test, (Activity)actIdent.IdentifiedActivity);
+
+                return true;
+            }
+            catch (Exception ex)
+            {
+                result = "Unexpected error occurred- " + ex.Message;
+                Reporter.ToLog(eLogLevel.ERROR, "Failed to export the Activities Group to QC/ALM", ex);
+                return false;
+            }
+        }
+
+        private static bool ExportActivityAsTestStep(Test test, Activity activity)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change return type to 'void'; not a single caller uses the returned value.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)